### DISCUSSION TITLE
KPI Status Icons broke when i accepted the change for warning on prefacing measure with table

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1240] [🙋♀️TE3 Q&A] [answered]
Hi - I am moving to TE3 for SSAS 1500 level
I one model i have not added calc groups.
Made minor folder changes, saved, and everything worked fine.
I was modify each measures where TE3 was giving warning about prefacing measure with table.
All seemed to work fine when i click FORMAT EDITOR which auto removes the name.
What did break was KPI Status Icons broke when i accepted the change for warning on prefacing measure with table, and saved.
I tried to add the table name back in and save but that did not seem to fix it.

I have a DEV copy of the same model where i didn't make this change in TE3.  I am comparing both broke and non broke in separate TE instances, and i cant see any differences in the properties.

Any ideas why this  would break and how to fix it?

**Comment by otykier** [comment]
If the icons appear differently, there must be some difference between the two versions of the model. Could you try to copy the measure from TE into a text editor, and examine the raw TMSL/JSON, to see if there are any differences?

**Comment by TinyTimothy2** [comment]
Of note the, VALUE part of the KPI still works fine.  It is just the icon.
Also of note the DISPLAY FOLDER is not what i would expect it to be. It has a double back slash for when the actual folder path in the GUI only has one.
 "displayFolder": "__Measures\\_Key Metrics - New",
[KPI Status Icon Broke.txt](https://github.com/TabularEditor/TabularEditor3/files/14422435/KPI.Status.Icon.Broke.txt)

**Reply by otykier** [reply]
Thank you for sending these details. It seems we have an unfortunate bug in our code formatter. Your original DAX for the KPI Status Expression looked like this:

```dax
var x='Fact Order Med'[KPI Burden Rate 2023]/'Fact Order Med'[_KPI Burden Rate 2023 Goal]
return
    if(ISBLANK(x),BLANK(),
        If(x<0.995,1,
            If(x<1.005,0,-1)
        )
    )            
```

But after formatting, it looks like this:

```dax
VAR x = [KPI Burden Rate 2023] / [KPI Burden Rate 2023]
RETURN
    IF( ISBLANK( x ), BLANK( ), IF( x < 0.995, 1, IF( x < 1.005, 0, -1 ) ) )
```

The problem here is that the reference to [_KPI Burden Rate 2023 Goal] was changed to [KPI Burden Rate 2023]. This is a different calculation than what you originally had, so this could explain the new icon you're seeing. We'll prioritize a fix for this bug with our next update - thanks!

**Reply by TinyTimothy2** [reply]
thx - I assume i can go back and manually change it.
Is this this just affecting KPIs? Or are there other measures we should be concerned about?

**Reply by TinyTimothy2** [reply]
FYI manually changing it fixed.

**Reply by otykier** [reply, marked as answer]
> Is this this just affecting KPIs? Or are there other measures we should be concerned about?

Only *references* to KPI expressions are affected, as far as we can tell.

**Reply by otykier** [reply]
@TinyTimothy2 FYI this issue was fixed in [3.15.0](https://docs.tabulareditor.com/te3/other/release-notes/3_15_0.html?tabs=downloads).
